Simplify parallel

pakrym-oai opened the pull request:
make tool processing return a future and then collect futures.
handle cleanup on Drop

pakrym-oai commented:
@codex give a thorough review, focus on lifetimes of tools call invocations

jif-oai commented on codex-rs/core/src/model_family.rs:
```diff
@@ -145,6 +146,8 @@ pub fn find_family_for_model(mut slug: &str) -> Option<ModelFamily> {
             reasoning_summary_format: ReasoningSummaryFormat::Experimental,
             base_instructions: GPT_5_CODEX_INSTRUCTIONS.to_string(),
             apply_patch_tool_type: Some(ApplyPatchToolType::Freeform),
+            experimental_supported_tools: vec!["read_file".to_string()],
```
To revert

jif-oai reviewed (commented):
Much cleaner with the handled moved in the async block

pakrym-oai committed:
core: refactor tool call handling for better parallelism

- Replace manual pending call tracking with FuturesOrdered for response processing
- Use RwLock and abort-on-drop for tool parallelization, supporting parallel and serial modes
- Switch tokio-util dependency to use "rt" feature
- Cleanup error handling and streamline handling of tool and non-tool response items

jif-oai reviewed (approved):
Ok once the comments are fixed

pakrym-oai committed:
fb

pakrym-oai committed:
fb
